wormhole-connect/src/routes/tbtc/tbtc.ts

```diff
@@ -0,0 +1,663 @@
+          message.vaa,
+        );
+        // We increase the gas limit estimation here by a factor of 10% to account for some faulty public JSON-RPC endpoints.
+        const gasLimit = estimateGas.mul(1100).div(1000);
+        const overrides = {
+          gasLimit,
+          // We use the legacy tx envelope here to avoid triggering gas price autodetection using EIP1559 for polygon.
+          // EIP1559 is not actually implemented in polygon. The node is only API compatible but this breaks some clients
+          // like ethers when choosing fees automatically.
+          ...(destChain === CHAIN_ID_POLYGON && { type: 0 }),
+        };
+        const tx = await L2WormholeGateway.receiveTbtc(message.vaa, overrides);
+        const receipt = await tx.wait();
+        const txId = await signAndSendTransaction(
+          destChainName,
+          receipt,
+          TransferWallet.RECEIVING,
+        );
+        return txId;
+      } else {
+        // If the gateway contract doesn't exist, then redeem with the token bridge
+        const tx = await wh.redeem(destChain, arrayify(message.vaa), undefined);
+        const txId = await signAndSendTransaction(
+          destChainName,
+          tx,
+          TransferWallet.RECEIVING,
+        );
+        return txId;
+      }
+    } else if (destChainId === CHAIN_ID_SOLANA) {
+      const signedVaa = Buffer.from(
+        arrayify(message.vaa, { allowMissingPrefix: true }),
+      );
+      const context = wh.getContext(
+        destChain,
+      ) as SolanaContext<WormholeContext>;
+      const connection = context.connection;
+      if (!connection) {
+        throw new Error('Connection not found');
+      }
+      const { core, token_bridge } = wh.mustGetContracts(destChain);
+      if (!core || !token_bridge) {
+        throw new Error('Core or token bridge not found');
+      }
+      await postVaa(connection, core, signedVaa);
+      const tx = await receiveTbtc(
+        signedVaa,
+        payer,
+        connection,
+        THRESHOLD_GATEWAYS[destChainId]!,
+        token_bridge,
+        core,
+      );
+      const txId = await signAndSendTransaction(
+        destChainName,
+        tx,
+        TransferWallet.RECEIVING,
+      );
+      return txId;
+    } else {
+      const context = wh.getContext(destChainId);
+      const tx = await context.redeem(
+        destChain,
+        arrayify(message.vaa),
+        undefined,
+        payer,
+      );
+      const txId = await signAndSendTransaction(
+        destChainName,
+        tx,
+        TransferWallet.RECEIVING,
+      );
+      return txId;
+    }
+  }
+
+  public async getPreview(
+    token: TokenConfig,
+    destToken: TokenConfig,
+    amount: number,
+    sendingChain: ChainName | ChainId,
+    receipientChain: ChainName | ChainId,
+    sendingGasEst: string,
+    claimingGasEst: string,
+    routeOptions?: any,
+  ): Promise<TransferDisplayData> {
+    const sendingChainName = wh.toChainName(sendingChain);
```

Comment:
Yeah, we should move the implementations of the `BridgeRoute`'s `getPreview`, `getTransferSourceInfo`, `getTransferDestInfo` methods to `RouteAbstract` and define those functions in `BridgeRoute` and `TBTCRoute` so that we use the base class implementation

Reply:
I think there's a `BaseRoute` which extends `RouteAbstract` and has some common method implementations, but not for these ones though

Reply:
Oh, good call. Makes sense to move those methods there instead of `RouteAbstract`